changelog.d/3347.change.rst

```diff
@@ -0,0 +1,3 @@
+Changed warnings and documentation notes about *experimental* aspect of ``pyproject.toml`` configuration:
+now ``[pyproject]`` is a fully supported configuration interface, but the ``[tool.setuptools]`` table
```

Comment:
FYI: This should be `[project]` not `[pyproject]`.

Reply:
Thank you @pradyunsg.

Reply:
Fixed in https://github.com/pypa/setuptools/commit/eaa4c0e3e51a91ce3447f4b82319e1679799acc2